clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -348,7 +368,9 @@ public synchronized void resume(Collection<TopicPartition> partitions) {
     public synchronized Map<TopicPartition, Long> endOffsets(Collection<TopicPartition> partitions) {
         Map<TopicPartition, Long> result = new HashMap<>();
         for (TopicPartition tp : partitions) {
-            Long endOffset = endOffsets.get(tp);
+            List<Long> offsets = endOffsets.get(tp);
+            offsets = offsets == null ? new ArrayList<Long>() : offsets;
+            Long endOffset = offsets.size() > 1 ? offsets.remove(0) : offsets.get(0);
```

Comment:
Doesn't this throw if `size() == 0` ? I think we need to check for this case and set `endOffset = null` explicitly?

Reply:
ack